React/Views/ScrollView/RCTScrollContentView.m

```diff
@@ -16,15 +16,36 @@ @implementation RCTScrollContentView
 
 - (void)reactSetFrame:(CGRect)frame
 {
-  [super reactSetFrame:frame];
-
 #if !TARGET_OS_OSX // TODO(macOS ISS#2323203)
   RCTScrollView *scrollView = (RCTScrollView *)self.superview.superview;
 #else // [TODO(macOS ISS#2323203)
   // macOS also has a NSClipView in its hierarchy
   RCTScrollView *scrollView = (RCTScrollView *)self.superview.superview.superview;
+
+  if (scrollView != nil) {
+    // On macOS scroll indicators may float over the content view like they do in iOS
+    // or depending on system preferences they may be outside of the content view
+    // which means the clip view will be smaller than the scroll view itself.
+    // In such cases the content view layout must shrink accordingly otherwise
+    // the contents will overflow causing the scroll indicators to appear unnecessarily.
+    NSScrollView *platformScrollView = scrollView.scrollView;
+    CGFloat verticalScrollerWidth = 0;
+    CGFloat horizontalScrollerHeight = 0;
+    if (platformScrollView.scrollerStyle == NSScrollerStyleLegacy) {
+      if (!platformScrollView.verticalScroller.isHidden) {
+        verticalScrollerWidth = platformScrollView.verticalScroller.frame.size.width;
```

Comment:
can save verticalScroller as an ivar to avoid redundant method invocations. This is a good example where dot notation makes that easy to be inefficient. Same below with horizontalScroller